### ISSUE TITLE
Failed to load resource: unsupported URL

### ISSUE THREAD

**Issue by fowlercraig** [roots/bud#771]
* [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [x] This request isn't a duplicate of an existing issue
* [x] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description

On a fresh install of the latest version of roots/sage, after running `yarn dev`, css & js assets aren't loading. They've been compiled and are present in the public directory, but aren't loading while viewing localhost:3000. 

## Steps to reproduce

1. `git clone https://github.com/roots/sage.git`
2. Update bud.config.js with target url (in this case, "http://test.local"
3. `composer install && yarn && yarn dev`
4. visit localhost:3000 once it's ready

**Expected behavior:** [What you expect to happen]

CSS and JS assets should load.

**Actual behavior:** [What actually happens]

CSS and JS assets no not load.

**Reproduces how often:** [What percentage of the time does it reproduce?]

100%

## Versions

roots/sage 10.0.0-beta.1: October 21st, 2021
DevKinsta Dev Suite (basically Docker with a frontend) Version 2.3.0 (2.3.0.2184)

## Additional information

Looking at the source, it appears that the asset urls are being output as
```
<script src="localhost:3000/wp-content/themes/sage-652b7ae24ca422d61503483adfb9518d27696b26/public/runtime.a5b3e3.js" id="app/0-js"></script>
<script src="localhost:3000/wp-content/themes/sage-652b7ae24ca422d61503483adfb9518d27696b26/public/app.c27b30.js" id="app/1-js"></script>
```
while in previous versions before roots/bud, they were output as
```
<script src="//localhost:3000/wp-content/themes/sage-652b7ae24ca422d61503483adfb9518d27696b26/public/runtime.a5b3e3.js" id="app/0-js"></script>
<script src="//localhost:3000/wp-content/themes/sage-652b7ae24ca422d61503483adfb9518d27696b26/public/app.c27b30.js" id="app/1-js"></script>
```

Tested on the laravel mix version of Sage and everything works as it should.

**Comment by kellymears** [comment]
@fowlercraig thanks for filing your issue. 

do you mind posting the contents of `sage-{xyz}/public/entrypoints.json`? this seems like it may be a problem with sage/acorn's asset loader and not bud itself.

**Comment by fowlercraig** [comment]
Here you go:

```
{
  "app": {
    "js": [
      "runtime.74a81a.js",
      "app.fb9c77.js"
    ],
    "dependencies": []
  },
  "editor": {
    "js": [
      "runtime.74a81a.js",
      "editor.24edba.js"
    ],
    "dependencies": [
      "wp-edit-post",
      "wp-blocks"
    ]
  },
  "customizer": {
    "js": [
      "runtime.74a81a.js",
      "customizer.885c23.js"
    ],
    "dependencies": []
  }
}

```

**Comment by QWp6t** [comment]
Couple of questions:

1. Are you using Soil or any other plugin that messes with URLs, such as creating relative URLs?

2. What do you have in config/assets.php?

Particularly right here:

https://github.com/roots/sage/blob/master/config/assets.php#L33-L40


Just trying to figure out whether this is a bud issue or an acorn issue, and where there might be a gap in our support for various environments.

**Comment by fowlercraig** [comment]
@QWp6t 

No, a vanilla install of Wordpress and new installation of Sage.

Here's my config/assets.php

```
<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default Assets Manifest
    |--------------------------------------------------------------------------
    |
    | Here you may specify the default asset manifest that should be used.
    | The "theme" manifest is recommended as the default as it cedes ultimate
    | authority of your application's assets to the theme.
    |
    */

    'default' => 'theme',

    /*
    |--------------------------------------------------------------------------
    | Assets Manifests
    |--------------------------------------------------------------------------
    |
    | Manifests contain lists of assets that are referenced by static keys that
    | point to dynamic locations, such as a cache-busted location. We currently
    | support two types of manifest:
    |
    | assets: key-value pairs to match assets to their revved counterparts
    |
    | bundles: a series of entrypoints for loading bundles
    |
    */

    'manifests' => [
        'theme' => [
            'path' => get_theme_file_path('public'),
            'url' => get_theme_file_uri('public'),
            'assets' => public_path('manifest.json'),
            'bundles' => public_path('entrypoints.json'),
        ]
    ]
];

```

Just a couple notes that will help you along the way:

- Vanilla install of Wordpress using DevKinsta/Docker
- Vanilla install of Sage
- New Mac M1
- PHP 8 (Sage says it only requires PHP 7.3, but I kept getting an error about a depencency requiring 8, so I upgraded)
- Noticing now that changes to HTML aren't reflected either

**Comment by fir-theme** [comment]
Same as above, but using Valet instead.
`get_theme_file_uri()` returns localhost:300, without any scheme or //.

updating `host: 'localhost'` >> `host: '//localhost'` seems to work, but I have no idea what other issues it might cause:
@roots>bud>lib>cjs>config.js
```
    server: {
        watch: {
            files: [],
        },
        middleware: {
            dev: true,
            hot: true,
            proxy: false,
        },
        browser: {
            indicator: true,
            overlay: true,
            log: true,
        },
        proxy: {
            target: 'localhost:8080',
        },
        host: 'localhost',
        port: 3000,
        methods: ['GET', 'HEAD'],
    },
```
This was the highest up the chain I could find to affect the data it compiles into the hmr.json file

**Comment by kellymears** [comment]
i was able to reproduce this and it should be resolved by #776. 

the problem is exactly as @fir-theme described: the current proxied host is set as `localhost` (rather than `[protcol]//localhost`.

the fix will involve a change to `bud.config.js` as the web path to theme assets will now need to be specified (`bud.setPublicPath('/app/themes/sage/public')` or the equivalent for your setup).

**Comment by kellymears** [comment]
I think this can be closed with the release of 5.0.0 which includes the fixes from #776 and works well for me in development.
